From a6eb67c58f6df25f9a5ac2717215497dfba577d6 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 22 Nov 2022 22:06:25 +0000
Subject: [PATCH 2/5] gnu: Add cl-assertion-error.

* gnu/packages/lisp-check.scm (cl-assertion-error, ecl-assertion-error,
  sbcl-assertion-error): New variables.
---
 gnu/packages/lisp-check.scm | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/gnu/packages/lisp-check.scm b/gnu/packages/lisp-check.scm
index 59033fdbc5..2d447533d9 100644
--- a/gnu/packages/lisp-check.scm
+++ b/gnu/packages/lisp-check.scm
@@ -94,6 +94,35 @@ (define-public cl-2am
 (define-public ecl-2am
   (sbcl-package->ecl-package sbcl-2am))
 
+(define-public sbcl-assertion-error
+  (let ((commit "8eab692a990d4caa193a46bae99af3e13e717b86")
+        (revision "1"))
+    (package
+      (name "sbcl-assertion-error")
+      (version (git-version "0.1.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/noloop/assertion-error")
+               (commit commit)))
+         (file-name (git-file-name "assertion-error" version))
+         (sha256
+          (base32 "0ix23kkakmf4nwx852zsssb831jvajr3qyppqfyks7y1ls617svn"))))
+      (build-system asdf-build-system/sbcl)
+      (inputs (list sbcl-dissect))
+      (home-page "https://github.com/noloop/assertion-error")
+      (synopsis "Error pattern for assertion libraries in Common Lisp")
+      (description "This package provides a Common Lisp assertion system with
+minimal dependencies on DISSECT.")
+      (license license:gpl3))))
+
+(define-public ecl-assertion-error
+  (sbcl-package->ecl-package sbcl-assertion-error))
+
+(define-public cl-assertion-error
+  (sbcl-package->cl-source-package sbcl-assertion-error))
+
 (define-public sbcl-check-it
   (let ((commit "b79c9103665be3976915b56b570038f03486e62f"))
     (package
-- 
2.38.0

